lib/widgets/components/purchase/purchase_overlay.dart

```diff
@@ -1,26 +1,23 @@
 import 'package:coffeecard/base/style/colors.dart';
-import 'package:coffeecard/cubits/environment/environment_cubit.dart';
 import 'package:coffeecard/cubits/purchase/purchase_cubit.dart';
 import 'package:coffeecard/cubits/receipt/receipt_cubit.dart';
 import 'package:coffeecard/cubits/tickets/tickets_cubit.dart';
-import 'package:coffeecard/models/receipts/receipt.dart';
+import 'package:coffeecard/models/purchase/payment.dart';
 import 'package:coffeecard/models/ticket/product.dart';
 import 'package:coffeecard/payment/payment_handler.dart';
 import 'package:coffeecard/service_locator.dart';
 import 'package:coffeecard/widgets/components/purchase/purchase_process.dart';
-import 'package:coffeecard/widgets/components/receipt/receipt_overlay.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 
 class PurchaseOverlay {
   final BuildContext _context;
 
-  void hide() {
-    Navigator.of(_context, rootNavigator: true).pop();
-  }
-
-  void show(InternalPaymentType paymentType, Product product) {
-    showDialog(
+  Future<Payment?> show(
```

Comment:
I like the changes so far.

Consider if this class with a method should be refactored into a function `showPurchaseOverlay(context, paymentType, product)` instead. Currently it holds no state and I don't see a reason for it to stay as a class.

See Material functions showDialog, showModalBottomSheet, and our own function appDialog.

Reply:
Well, it could be refactored like that easily. I think the main argument for not doing so is that our over overlays work in this manner instead, i.e. the receipt and loading overlays. Mind you, for those two it is also to provide ourselves with the convenience of the hide method, so we always use the right navigator. That argument no longer applies to this class, so I would be fine with refactoring, if you still prefer that.

Reply:
Yes I would prefer refactoring into a function.

For all three overlays we want to use the root navigator, which is always accessible with `Navigator.of(context, rootNavigator: true)` and `showDialog(useRootNavigator: true, [other args...]);`. So I think that the other two classes also need to be refactored, but that can be looked at later.